### PR TITLE
[CMake] Add doxygen targets for documentation

### PR DESCRIPTION
**Pull request by feliwir**
Eventually we want to add a MainPage.dox at some point for some more detailed documentation

### CODE REVIEW

File: GeneralsMD/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,14 @@ add_feature_info(ZeroHourTools GENZH_BUILD_ZEROHOUR_TOOLS "Build Zero Hour Mod T
 option(GENZH_BUILD_ZEROHOUR_EXTRAS "Build extra tools/tests for Zero Hour" OFF)
 add_feature_info(ZeroHourExtras GENZH_BUILD_ZEROHOUR_EXTRAS "Build Zero Hour Extra Tools/Tests")
 
+# Do we want to build documentation?
+option(GENZH_BUILD_ZEROHOUR_DOCS "Build documentation for Zero Hour" OFF)
+add_feature_info(GeneralsDocs GENZH_BUILD_ZEROHOUR_DOCS "Build Zero Hour Documentation")
```
Comment (xezon) [inline GeneralsMD/CMakeLists.txt]: This looks like it should be `ZeroHourDocs`

File: Generals/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,14 @@ add_feature_info(GeneralsTools GENZH_BUILD_GENERALS_TOOLS "Build Generals Mod To
 option(GENZH_BUILD_GENERALS_EXTRAS "Build extra tools/tests for Generals" OFF)
 add_feature_info(GeneralsExtras GENZH_BUILD_GENERALS_EXTRAS "Build Generals Extra Tools/Tests")
 
+# Do we want to build documentation?
+option(GENZH_BUILD_GENERALS_DOCS "Build documentation for Generals" OFF)
```
Comment (xezon) [inline Generals/CMakeLists.txt:12]: This option does not show in cmake. `GENZH_BUILD_ZEROHOUR_DOCS` does. Perhaps because of the other error.

Comment (feliwir) [reply Generals/CMakeLists.txt:12]: Should be shown when BUILD_GENERALS is enabled.

File: GeneralsMD/CMakeLists.txt (hunk 1 above)
Comment (feliwir) [reply GeneralsMD/CMakeLists.txt]: Fixed

File: Generals/CMakeLists.txt (hunk 2 above)
Comment (xezon) [reply Generals/CMakeLists.txt:12]: Ok. It does show then yes.

**Comment by xezon** [comment]
I get cmake error on local configure:

```
CMake Error at C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
Call Stack (most recent call first):
  C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files (x86)/cmake/cmake-3.30.4-windows-x86_64/share/cmake-3.30/Modules/FindDoxygen.cmake:681 (find_package_handle_standard_args)
  GeneralsMD/CMakeLists.txt:15 (find_package)
```

Perhaps needs to be turned off if DOXYGEN_EXECUTABLE is not present?

**Comment by xezon** [comment]
Ok that makes sense yes.

**Review by xezon — APPROVED** [review]
